servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -10589,6 +10607,16 @@ boolean isAllowedDeletePendingMembership(Principal principal, final String domai
         return pendingMember != null && principal.getFullName().equals(pendingMember.getRequestPrincipal());
     }
 
+    boolean isAllowedDeleteAssertion(Principal principal, final AthenzDomain domain, final String policyName, final Long assertionId) {
+          if (hasAccess(domain, "update", String.format("%s:%s%s", domain.getName(), POLICY_PREFIX, policyName), principal, null) == AccessStatus.ALLOWED) {
+              return true;
+          }
+          if (hasAccess(domain, "delete", String.format("%s:%s%d", domain.getName(), ASSERTION_PREFIX, assertionId), principal, null) == AccessStatus.ALLOWED) {
```

Comment:
this is the wrong authorization for the command. assertion is not a top level object within the domain and there is no requirement that the id is unique across the domain. It happens to be that way with our implementation but someone else might define their own store implementation and only support unique assertion ids per policy. The authorization would be 

authorize("delete", "{domainName}:policy.{policyName}.assertion.{assertionId}" )